presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileWriter.java

```diff
@@ -194,10 +195,12 @@ public void close()
             output.close();
         }
         finally {
-            keySectionOutput.destroy();
+            Closer closer = Closer.create();
```

Comment:
Consider `try (Closer closer = ...)` if you already use `Closer`.

Also, this `Closer` instance could also handle the `try` section, eliminating `try-finally`. 
(Mind the order in which closer closes though).